android/source/VideoLocker/src/org/edx/mobile/module/analytics/SegmentTracker.java

```diff
@@ -13,27 +13,31 @@
 import com.segment.analytics.Traits;
 
 public class SegmentTracker {
-    
+
+    /* Singleton instance of Analytics */
     private Analytics analytics;
     
     public SegmentTracker(Context context) {
         try {
             String writeKey = Environment.getInstance().getConfig().getSegmentIOWriteKey();
-            String debugging = context.getString(R.string.analytics_debug);
-            int queueSize = context.getResources().getInteger(R.integer.analytics_queue_size);
 
-            if(writeKey!=null) {
+            if (writeKey != null) {
+                String debugging = context.getString(R.string.analytics_debug);
+                int queueSize = context.getResources().getInteger(R.integer.analytics_queue_size);
+
                 LogUtil.log(getClass().getName(), "SegmentTracker created with write key: " + writeKey);
                 // Now Analytics.with will return the custom instance
                 analytics = new Analytics.Builder(context, writeKey)
                         .debugging(Boolean.parseBoolean(debugging))
                         .queueSize(queueSize)
                         .build();
+            } else {
+                LogUtil.error(getClass().getName(), "writeKey is null, Segment analytics will not work.");
```

Comment:
@nasthagiri  I added a log when <code>writeKey</code> is null.  
Yes, the production configuration has a <code>writeKey</code> and we will always ensure it. Also, these changes will allow any configuration to work even without a <code>writeKey</code>